manifests/client.pp

```diff
@@ -77,9 +77,12 @@
       ensure => $package_ensure,
       name   => $package_name,
     }
-    $req = Package['snmp-client']
-  } else {
-    $req = undef
+  }
+
+  $req = $::osfamily ? {
+    'RedHat' => [Package['snmp-client'], Package[$snmp::params::package_name]],
```

Comment:
There's also a bit of a logic flaw in the osfamily selector.  You may want to unmerge this commit until I can get things fixed up.